### DISCUSSION TITLE
Can't generate proper svg (php 5)

### DISCUSSION THREAD

**Discussion by AlexisHTC** [chillerlan/php-qrcode#129] [Q&A]
- When i try to generate svg , I can't get finder and data in the same svg, and I don't get it ,  I know the php version is old but it's not my project 

**Code sample**
```php
 public function gn_qrcode(){
        $options = new QROptions([
            'version'             => 7,
            'outputType'          => QRCode::OUTPUT_MARKUP_SVG,
            'imageBase64'         => false,
            'moduleValues' => [
                1024 => 'url(#g2)',
                4    => '#000',
                6    => '#eee', // light (false)
                // finder
                12   => '#eee',
                14   => '#eee',
                // alignment
                10   => '#eee',
                1536 => 'url(#g1)', // dark (true)
                2560 => 'url(#g1)',
                // timing
                3072 => 'url(#g1)',
                // format
                3584 => 'url(#g1)',
                16   => '#eee',
                // version
                4096 => 'url(#g1)',
                // data
                // darkmodule
                512  => 'url(#g1)',
                // separator
                8    => '#eee',
                // quietzone
                18   => '#eee',
            ],
        ]);


        $qr = (new QRCode($options))->render(urlencode('url'));
        return $qr;
    }
```

**Screenshots**

![image](https://user-images.githubusercontent.com/105217087/172407315-a0de68e8-9e56-429f-baf6-db63f2be574a.png)
![image](https://user-images.githubusercontent.com/105217087/172407359-5af4d38a-6aa6-4585-be36-83a7b076b186.png)

 - PHP version 5.6.40
 - Library version:  "chillerlan/php-qrcode": "^1.0"

**Comment by codemasher** [comment]
Hey, in v1 the dark values are always shifted by 8 bits left from the initial values, that can be found [over here in `QRMatrix`](https://github.com/chillerlan/php-qrcode/blob/00103824c313d4c8509992ce6369cc27b5f1b795/src/Data/QRMatrix.php#L24). So data should be `0x04` (4) and `0x04 << 8` (1024) respectively. The finder patterns are `0x06` (6) and `0x06 << 8` (1536). The values are are present in your options array but are a bit mixed up. However, it seems that the value for several patterns is invalid as the SVG `<defs>` with the `#g1` and `#g2` are missing from the options. Check the full [SVG example for v1 over here](https://github.com/chillerlan/php-qrcode/blob/00103824c313d4c8509992ce6369cc27b5f1b795/examples/svg.php#L29-L38).
